src/main/java/org/broadinstitute/hellbender/utils/downsampling/Downsampler.java

```diff
@@ -0,0 +1,123 @@
+package org.broadinstitute.hellbender.utils.downsampling;
+
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * The basic downsampler API, with no reads-specific operations.
+ *
+ * Downsamplers that extend this class rather than the ReadsDownsampler class can handle
+ * any kind of item, however they cannot be wrapped within a DownsamplingReadsIterator or a
+ * PerSampleDownsamplingReadsIterator.
```

Comment:
There are several comments scattered throughout this PR mentioning `DownsamplingReadsIterator` and/or `PerSampleDownsamplingReadsIterator`, but these classes have not been ported. Should either remove these comments or port the iterators :)